### DISCUSSION TITLE
Get a Node without using a NodeID

### DISCUSSION THREAD

**Discussion by ingdesarr1** [FreeOpcUa/opcua-asyncio#1616] [Q&A]
Good afternoon. In this moment, I'm trying to use asyncua with a Siemens PLC S7-1200. This PLC does not use the NodeID as a String but as an integer, and it's more difficult to identify the variables to work with. I was thinking of using the DisplayName to get the variables, but I don't know if it's possible. thank you

**Comment by schroeder-** [comment]
You can use the browse_name. You need to give the full path from the starting object.
For example if you want to start at the root  of the server: 

```
var = await client.nodes.root.get_child(
            f"0:Objects/{nsidx}:MyObject/{nsidx}:MyVariable"
 )
```

if you want to start from a specific node you can this:

```
node = client.get_node("ns=2;i=4")
var = await node.get_child(
            "{nsidx}:MyVariable"
 )
```
